src/components/Wowerlay.tsx

```diff
@@ -103,69 +123,97 @@ export const Wowerlay = defineComponent({
       window.removeEventListener('click', handleWindowClick);
     });
 
+    /**
+     * This variable is for watching Popover's animation state.
+     * When we unmount popover if user passed a transition we need to wait
+     * for transition to end to hide Popover.
+     *
+     * When popover animation ends this variable will hide background.
+     */
     const backgroundVisible = ref(props.visible);
 
+    /**
+     * Hides background element (so popover) when Popover animation ends.
+     */
+    const handleContentTransitionEnd = () => {
+      backgroundVisible.value = false;
+    };
+
     watch(
       () => props.visible,
       (state) => {
         if (state) {
+          // We need to prevent closing popover in the same event loop because
+          // If users clicks a button and changes visible to true
+          // Wowerlay will be opened but same click event may be triggered on window
+          // due to bubbling so it will be cached by our Window:click listener.
```

Comment:
```suggestion
// Use setTimeout to prevent immediate popover closure caused by the same click event due to event bubbling.
// We listen to the window, so this ensures a single click doesn't simultaneously open and close the popover.
```